src/models/hooks/sessionReportPilot.js

```diff
@@ -213,6 +213,76 @@ export const createGoalsForSessionRecipientsIfNecessary = async (sequelize, sess
   }
 };
 
+export const removeGoalsForSessionRecipientsIfNecessary = async (sequelize, sessionReportOrInstance, options) => {
+  const processSessionReport = async (sessionReport) => {
+    let event;
+    let nextSessionRecipients;
+
+    if (sessionReport && sessionReport.data) {
+      const data = (typeof sessionReport.data.val === 'string')
+        ? JSON.parse(sessionReport.data.val)
+        : sessionReport.data;
+
+      event = data.event;
+      nextSessionRecipients = data.recipients;
+    }
+
+    if (!event.id || !sessionReport.id) return;
+
+    nextSessionRecipients = nextSessionRecipients.map((r) => r.value);
+
+    const recipients = await sequelize.models.EventReportPilotGoal.findAll({
+      where: { eventId: event.id, sessionId: sessionReport.id },
+      transaction: options.transaction,
+      raw: true,
+    });
+
+    if (!recipients || !recipients.length) return;
+
+    // Using the EventReportPilotGoal table, find all goals that were created for this recipient (grantId).
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const entry of recipients) {
+      if (!nextSessionRecipients.includes(entry.grantId)) {
+        // A recipient was removed.
+
+        // If the goal is referenced in an ARG, then it's in use and cannot be deleted.
+        // This should _never_ be possible, because this hook happens when a session is edited.
+        // If the session is edited, it's not complete, and therefore the goal should not be in use.
+        const args = await sequelize.models.ActivityReportGoal.findAll({
```

Comment:
Nitpick args = arguments, maybe activityReportGoals.

Reply:
Yeah.. it's activity report goals. I can make that more clear by renaming it to `activityReportGoals`, lol.